libs/blocks/marketo/marketo.js

```diff
@@ -18,13 +18,32 @@ import { parseEncodedConfig, loadScript, createTag } from '../../utils/utils.js'
 const FORM_ID = 'form id';
 const BASE_URL = 'marketo host';
 const MUNCHKIN_ID = 'marketo munckin';
+const FORM_MAP = {
+  'destination-url': 'form.success.content',
+  'co-partner-names': 'program.copartnernames',
+  'sfdc-campaign-id': 'program.campaignids.sfdc',
+};
 
 export const formValidate = (form) => {
   const formEl = form.getFormElem().get(0);
   formEl.classList.remove('hide-errors');
   formEl.classList.add('show-warnings');
 };
 
+export const decorateURL = (destination, baseURL = window.location) => {
+  let destinationUrl = new URL(destination, baseURL.origin);
```

Comment:
Is destination guaranteed to be a valid url?  If it isn't this will throw

Reply:
Good call, refactored this to a try/catch and made sure we have a hostname, added a test case to validate.